build_tools/cmake/iree_cc_library.cmake

```diff
@@ -6,6 +6,35 @@
 
 include(CMakeParseArguments)
 
+# iree_filter_cc_deps(DEPS_VAR)
+#
+# Filters a list of CC dependencies, making alterations as needed.
+# At present, this is used to redirect LLVM libraries to the libLLVM.so
+# dynamic library, when LLVM has been configured to link against it.
+# This is necessary to preserve the one-definition rule in the build graph
+# in a consistent way as to how AddLLVM.cmake does it.
+# Note that this only really works if libLLVM.so was configured to contain
+# "all" components. If this ever becomes unworkable, we may need to port the
+# component naming logic and selectively choose when to divert.
+function(iree_filter_cc_deps DEPS_VAR)
+  set(_deps ${${DEPS_VAR}})
+  set(_new_deps)
+  set(_modified FALSE)
+  foreach(_dep ${_deps})
+    # If linking against the LLVM dylib, then divert any LLVM prefixed
+    # targets there.
+    if(LLVM_LINK_LLVM_DYLIB AND _dep MATCHES "^LLVM")
+      list(APPEND _new_deps "LLVM")
```

Comment:
This is potentially adding `"LLVM"` multiple times.  Maybe just at the end, before writing to `PARENT_SCOPE`, do a `list(REMOVE_DUPLICATES ${DEPS_VAR})` ?

Reply:
I think I'm going to leave it. CMake properly dedupes the list when forming linker command lines, and I have sometimes found it useful to not "topographically disrupt" things when debugging gotchas. With the duplicates in place, it becomes possible to scan the resultant list and have a good guess of which library caused the dep.